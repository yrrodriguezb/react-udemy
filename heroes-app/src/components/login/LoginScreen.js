import React, { useContext } from 'react';
import { useNavigate } from 'react-router-dom';
import { AuthContext } from '../../auth/authContext';
import { types } from '../../types/types';

export const LoginScreen = () => {

  const navigate = useNavigate();
  const { dispatch } = useContext(AuthContext);

  const login = () => {
    const action = {
      type: types.login,
      payload: {
        name: 'Yeison Rodriguez'
      }
    }

    dispatch(action);

    const lastPath = localStorage.getItem("lastPath") || "/marvel"

    navigate(lastPath, {
      replace: true
    });
  }

  return (
    <div className='container mt-5'>
      <h1>Login</h1>
      <hr />

      <button
        className='btn btn-primary'
        onClick={ login }
      >
        Login
      </button>
    </div>
  )
}
