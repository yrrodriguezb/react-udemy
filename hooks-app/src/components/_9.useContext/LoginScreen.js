import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

  const { setUser } = useContext(UserContext);

  const login = () => {
    setUser({
      id: 123,
      name: 'Yeison Rodriguez',
      email: 'yrrodriguezb@gmail.com',
      login_at: new Date().getTime()
    })
  }

  return (
    <div className="p-5">
      <h1>LoginScreen</h1>
      <hr />

      <button 
        type="button" 
        className="btn btn-outline-primary"
        onClick={ login }
      >
        Login
      </button>
    
    </div>
  )
}
