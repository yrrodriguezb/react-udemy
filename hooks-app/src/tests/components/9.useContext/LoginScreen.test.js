import React from 'react';
import { mount } from "enzyme";
import '@testing-library/jest-dom'
import { LoginScreen } from '../../../components/_9.useContext/LoginScreen';
import { UserContext } from '../../../components/_9.useContext/UserContext';


describe('Pruebas en <LoginScreen />', () => {
  const setUser = jest.fn();
  
  const wrapper = mount(
    <UserContext.Provider value={ { setUser } }>
      <LoginScreen />
    </UserContext.Provider>
  );

  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe ejectar el setUser con el argumento esperado', () => {
    wrapper.find('button').prop('onClick')();

    expect(setUser).toHaveBeenCalledWith({
      id: 123,
      name: 'Yeison Rodriguez',
      email: 'yrrodriguezb@gmail.com',
      login_at: expect.any(Number)
    });
  })
})