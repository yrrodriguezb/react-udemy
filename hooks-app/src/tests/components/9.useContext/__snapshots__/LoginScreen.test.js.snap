// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> Debe mostrarse correctamente 1`] = `
<div
  className="p-5"
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
    type="button"
  >
    Login
  </button>
</div>
`;
